src/core/diags/factory_diags.cpp

```diff
@@ -657,31 +657,81 @@ Error Diags::ProcessRadio(uint8_t aArgsLength, char *aArgs[])
             ExitNow();
         }
 
-        if (StringMatch(aArgs[0], "async"))
+        if (StringMatch(aArgs[0], "filter"))
         {
             aArgs++;
             aArgsLength--;
-            receiveConfig.mIsAsyncCommand = true;
-        }
 
-        VerifyOrExit(aArgsLength > 0);
-        SuccessOrExit(error = Utils::CmdLineParser::ParseAsUint16(aArgs[0], receiveConfig.mNumFrames));
-        aArgs++;
-        aArgsLength--;
+            VerifyOrExit(aArgsLength > 0);
 
-        if (aArgsLength > 0)
-        {
-            SuccessOrExit(error = ParseReceiveConfigFormat(aArgs[0], receiveConfig));
+            if (StringMatch(aArgs[0], "enable"))
+            {
+                mReceiveConfig.mIsFilterEnabled = true;
+                error                           = kErrorNone;
+            }
+            else if (StringMatch(aArgs[0], "disable"))
+            {
+                mReceiveConfig.mIsFilterEnabled = false;
+                error                           = kErrorNone;
+            }
+            else
+            {
+                Mac::Address dstAddress;
+
+                if (StringMatch(aArgs[0], "-"))
+                {
+                    dstAddress.SetNone();
+                    error = kErrorNone;
+                }
+                else if (strlen(aArgs[0]) == 2 * sizeof(Mac::ExtAddress))
+                {
+                    Mac::ExtAddress extAddress;
+
+                    SuccessOrExit(error = Utils::CmdLineParser::ParseAsHexString(aArgs[0], extAddress.m8));
+                    mReceiveConfig.mFilterAddress.SetExtended(extAddress);
+                }
+                else
+                {
+                    Mac::ShortAddress shortAddress;
+
+                    SuccessOrExit(error = Utils::CmdLineParser::ParseAsUint16(aArgs[0], shortAddress));
+                    mReceiveConfig.mFilterAddress.SetShort(shortAddress);
+                }
+            }
```

Comment:
The code looks good, but `git diff` makes it look like a larger change. I guess this is due to the addition of `else` and indentation of the next block (related to "async" checks).

Can we consider adding `ExitNow()` at the end of the `if (StringMatch(aArgs[0], "filter"))` block? This way, we can keep the `if (StringMatch(aArgs[0], "async"))` section as is (no need to place it in `else` and indent).